### PR TITLE
prevent the creation of nodepools for deleted clusters

### PR DESCRIPTION
**Pull request by anvddriesch**
Towards: https://github.com/giantswarm/giantswarm/issues/11174

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [![CircleCI](https://circleci.com/gh/giantswarm/aws-admission-controller.svg?style=shield)](https://circleci.com/gh/giantswarm/aws-admission-controller)
 
-# G8S Admission Controller
+# AWS Admission Controller
```
Comment (anvddriesch) [inline README.md:3]: I just figured this has to be updated

**Review by calvix — APPROVED** [review]

File: pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go [hunk 2]
```diff
@@ -68,6 +78,38 @@ func (v *Validator) Validate(request *admissionv1.AdmissionRequest) (bool, error
...
+	err = v.ValidateCluster(awsMachineDeployment)
+	if err != nil {
+		return false, microerror.Mask(err)
+
+	}
+
+	err = v.MachineDeploymentLabelMatch(awsMachineDeployment)
+	if err != nil {
+		return false, microerror.Mask(err)
+
```
Comment (njuettner) [inline pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go:97]:
```suggestion
```

File: pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go [hunk 3]
```diff
@@ -155,6 +197,26 @@ func (v *Validator) MachineDeploymentAnnotationPauseTimeIsValid(awsMachineDeploy
...
+	var err error
+
+	// Retrieve the `Cluster` CR related to this object.
+	cluster, err := aws.FetchCluster(&aws.Handler{K8sClient: v.k8sClient, Logger: v.logger}, &awsMachineDeployment)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	// make sure the cluster is not deleted
+	if cluster.DeletionTimestamp != nil {
+		v.logger.Log("level", "debug", "message", fmt.Sprintf("AWSMachineDeployment could not be created because Cluster '%s' is in deleting state.",
```
Comment (njuettner) [inline pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go:210]: I think this should be removed, either we log or we return the error.

File: pkg/aws/machinedeployment/validate_machinedeployment.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	var err error
+
+	// Retrieve the `Cluster` CR related to this object.
+	cluster, err := aws.FetchCluster(&aws.Handler{K8sClient: v.k8sClient, Logger: v.logger}, &machineDeployment)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	// make sure the cluster is not deleted
+	if cluster.DeletionTimestamp != nil {
+		v.logger.Log("level", "debug", "message", fmt.Sprintf("MachineDeployment could not be created because Cluster '%s' is in deleting state.",
```
Comment (njuettner) [inline pkg/aws/machinedeployment/validate_machinedeployment.go:70]: same here

File: pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go (hunk 3 above)
Comment (anvddriesch) [reply pkg/aws/awsmachinedeployment/validate_awsmachinedeployment.go:209]: Makes sense. In that case I want to make sure that the error appears in the logs after validation failed.

**Pull request merged**